src/components/VaultCharts.tsx

```diff
@@ -3,32 +3,41 @@ import useSWR from 'swr/immutable';
 import { VaultTotalLockedCollateralChart } from '@/widgets/VaultTotalLockedCollateralChart';
 import { VaultTotalMintedISTChart } from '@/widgets/VaultTotalMintedISTChart';
 import { VAULTS_DAILY_METRICS_QUERY } from '@/queries';
-import { extractDailyOracles, populateMissingDays, subQueryFetcher } from '@/utils';
+import { createNumberWithLeadingZeroes, extractDailyOracles, populateMissingDays, subQueryFetcher } from '@/utils';
 import ChartsSkeleton from './ChartsSkeleton';
 import { GRAPH_DAYS } from '@/constants';
 import { DailyOracles, FormattedGraphData } from '@/types/common';
 import { GraphData } from '@/types/psm-types';
-import { VaultDailyMetricsQueryResponse } from '@/types/vault-types';
+import { BoardAuxesMap, VaultDailyMetricsQueryResponse, VaultManagerMetricsNode } from '@/types/vault-types';
 
 type Props = {
   tokenNames: Array<string>;
   vaultsDataIsLoading: boolean;
+  boardAuxes: BoardAuxesMap;
   error: any;
 };
 
-export function populateGraphData(dailyOracles: DailyOracles, nodes: any[], graphData: Record<string, GraphData>): void {
+export function populateGraphData(
+  dailyOracles: DailyOracles,
+  nodes: any,
+  boardAuxes: BoardAuxesMap,
+  graphData: Record<string, GraphData>,
+): void {
   for (let j = 0; j < nodes?.length; j++) {
     const dailyTokenMetrics = nodes[j];
     const dateKey = dailyTokenMetrics?.dateKey;
 
     const oracle = (dailyOracles && dailyOracles[dateKey]) || { typeOutAmountLast: 1, typeInAmountLast: 1 };
     const blockTime = dailyTokenMetrics?.blockTimeLast?.slice(0, 10);
     const liquidatingCollateralBrand = dailyTokenMetrics?.liquidatingCollateralBrand;
-    const totalCollateralLast = dailyTokenMetrics?.totalCollateralLast;
-    const totalDebtLast = dailyTokenMetrics?.totalDebtLast / 1_000_000;
+    const decimalPlaces = (boardAuxes && boardAuxes[liquidatingCollateralBrand]) || 6;
+    const divisor = createNumberWithLeadingZeroes(decimalPlaces);
+
+    const totalCollateralLast = Number(dailyTokenMetrics?.totalCollateralLast);
+    const totalDebtLast = Number(dailyTokenMetrics?.totalDebtLast) / divisor;
```

Comment:
i dont think the divisor is used in this case

Reply:
since this field represents IST, and the divisor is only for other tokens like stATOM, stOSMO etc

Reply:
It was a recent change [over here](https://github.com/agoric-labs/agoric-inter-dashboard/commit/79ec8ad517f714ab7da53a024794fdfa95b6c447)